src/Groups/homomorphisms.jl

```diff
@@ -388,13 +388,40 @@ _get_iso_function(::Type{PcGroup}) = GAP.Globals.IsomorphismPcGroup
 
 Return an isomorphism from `G` to a group of type `T`.
 An exception is thrown if no such isomorphism exists.
+
+Isomorphisms are cached in `G`, subsequent calls of `isomorphism` with the
+same `T` yield identical results.
+
+If only the image of such an isomorphism is needed, use `T(G)`.
+
+# Examples
+```jldoctest
+julia> G = dihedral_group(6)
+<pc group of size 6 with 2 generators>
+
+julia> iso = isomorphism(PermGroup, G)
+Group homomorphism from
+<pc group of size 6 with 2 generators>
+to
+Group([ (1,2)(3,6)(4,5), (1,3,5)(2,4,6) ])
+
+julia> PermGroup(G)
+Group([ (1,2)(3,6)(4,5), (1,3,5)(2,4,6) ])
+
+julia> codomain(iso) === ans
+true
+```
 """
 function isomorphism(::Type{T}, G::GAPGroup) where T <: Union{FPGroup, PcGroup, PermGroup}
-   fun = _get_iso_function(T)
-   f = fun(G.X)::GapObj
-   f == GAP.Globals.fail && throw(ArgumentError("Could not convert group into a group of type $T"))
-   H = T(GAP.Globals.ImagesSource(f)::GapObj)
-   return GAPGroupHomomorphism(G, H, f)
+   # Known isomorphism are cached in the attribute `:isomorphisms`.
+   isos = get_attribute!(() -> Dict{Type, Any}(), G, :isomorphisms)
```

Comment:
I think we fixed `get_attribute! to take type names, so perhaps try if this now works:
```suggestion
   isos = get_attribute!(Dict{Type, Any}, G, :isomorphisms)
```